### PR TITLE
Add functionality to delete applications

### PR DESCRIPTION
**Pull request by JayShenoy**

### CODE REVIEW

**Comment by dcrankshaw** [comment]
@JayShenoy can you delete all the cmake files you committed? This PR has 13,000 files and over a million lines of code.

**Comment by JayShenoy** [comment]
My mistake, I just removed that mess

**Comment by JayShenoy** [comment]
@dcrankshaw The frontend and management tests now all pass, but the integration tests are failing because of issues with Docker that I believe are local to my machine.

**Comment by JayShenoy** [comment]
Jenkins test this please

**Comment by dcrankshaw** [comment]
jenkins add to whitelist

**Review by dcrankshaw — APPROVED** [review]
This looks great. Very nice work. I'll merge once the tests pass.

File: clipper_admin/clipper_admin/clipper_admin.py [hunk 1]
```diff
@@ -188,6 +188,25 @@ def register_application(self, name, input_type, default_output,
...
+        headers = {"Content-type": "application/json"}
+        r = requests.post(url, headers=headers, data=req_json)
+        logger.debug(r.text)
+        if r.status_code != requests.codes.ok:
+            msg = "Received error status code: {code} and message: {msg}".format(
+                code=r.status_code, msg=r.text)
+            logger.error(msg)
+            raise ClipperException(msg)
+        else:
+            logger.info("Application {app} was successfully deleted").format(
```
Comment (simon-mo) [inline clipper_admin/clipper_admin/clipper_admin.py]: This line will error:
```
    logger.info("Application {app} was successfully deleted").format(
AttributeError: 'NoneType' object has no attribute 'format'
```

File: integration-tests/clipper_admin_tests.py [hunk 2]
```diff
@@ -80,6 +80,18 @@ def test_link_not_registered_model_to_app_fails(self):
             self.clipper_conn.link_model_to_app(app_name, not_deployed_model)
         self.assertTrue("No model with name" in str(context.exception))
 
+    def test_delete_application_correct(self):
+        nput_type = "doubles"
+        default_output = "DEFAULT"
+        slo_micros = 30000
+        app_name = "testapp"
+        self.clipper_conn.register_application(app_name, input_type,
```
Comment (simon-mo) [inline integration-tests/clipper_admin_tests.py]: This line will error:
```
Traceback (most recent call last):
  File "clipper_admin_tests.py", line 88, in test_delete_application_correct
    self.clipper_conn.register_application(app_name, input_type,
NameError: global name 'input_type' is not defined
```

File: integration-tests/clipper_admin_tests.py [hunk 3]
```diff
@@ -80,6 +80,18 @@ def test_link_not_registered_model_to_app_fails(self):
             self.clipper_conn.link_model_to_app(app_name, not_deployed_model)
         self.assertTrue("No model with name" in str(context.exception))
 
+    def test_delete_application_correct(self):
```
Comment (simon-mo) [inline integration-tests/clipper_admin_tests.py:83]: You also need to add this test function to SHORT_TEST_ORDERING list below:
https://github.com/ucbrise/clipper/blob/ff849d4ade612732a72427b156206ec9333073fe/integration-tests/clipper_admin_tests.py#L567

**Review by simon-mo — COMMENTED** [review]
@JayShenoy Great work. A quick question: does this PR intend to just delete applications? Is it suppose to do any clean up between app_model links. 

For example, if I registered an application `simple-app`, and deploy model `sum-model` linking to `simple-app`, right now if I delete `simple-app`, the model container will not be deleted. This is intentional? 

Thanks for the great work!

**Comment by dcrankshaw** [comment]
@simon-mo Yeah that behavior is correct. Deleting an application should not delete any models.

File: integration-tests/clipper_admin_tests.py (hunk 3 above)
Comment (dcrankshaw) [reply integration-tests/clipper_admin_tests.py:83]: Oh good catch!

**Comment by JayShenoy** [comment]
@simon-mo This PR is purely intended to delete applications. I believe someone else is working on deleting models.

@dcrankshaw I have made the necessary revisions.

**Comment by JayShenoy** [comment]
Bless up 🙏

**Comment by dcrankshaw** [comment]
Nice job!

**Pull request merged**